discovery/sync_manager.go

```diff
@@ -398,6 +398,13 @@ func (m *SyncManager) syncerHandler() {
 				continue
 			}
 
+			// We may not even have enough inactive syncers to be
+			// transitted. In that case, we will transit all the
+			// inactive syncers.
+			if len(m.inactiveSyncers) < numActiveLeft {
+				numActiveLeft = len(m.inactiveSyncers)
```

Comment:
So before, it would try to rotate `numActiveLeft` times when there were less?

Reply:
Yes. From itest it's rotating 3 times no matter what.